File: 0011-container-with-most-water/0011-container-with-most-water.js
/**
 * @param {number[]} height
 * @return {number}
 */
var maxArea = function(height) {
    let mx = 0 , l  = 0 , r = height.length-1 
    while(l < r) {
        let len = Math.min(height[l] , height[r]) , width = r - l , area = len * width 

        mx = Math.max(area , mx)
        if(height[l] < height[r]) l++
        else r--
    }
    return mx 
};

File: 0011-container-with-most-water/0011-container-with-most-water.java
public class Solution {

    public int maxArea(int[] height) {
        int maxAreaFound = 0;
        int leftIndex = 0;
        int rightIndex = height.length - 1;

        while (leftIndex < rightIndex) {
            int currentWidth = rightIndex - leftIndex;
            int currentHeight = Math.min(height[leftIndex], height[rightIndex]);
            int currentArea = currentWidth * currentHeight;

            maxAreaFound = Math.max(maxAreaFound, currentArea);

            // Move the pointer of the shorter line.
            if (height[leftIndex] < height[rightIndex]) {

                // Moving left pointer for potentially larger height.
                leftIndex++;

            } else {

                // Moving the right pointer.
                rightIndex--;

            }
        }

        return maxAreaFound;
    }
}